test/JDBC/expected/alter_proc_recompile-vu-verify.out

```diff
@@ -186,12 +186,12 @@ text
 Query Text: EXEC    p_alter_recomp_02 0
   Query Text: select 'p_alter_recomp_02', * from tb_alter_recomp_1 where c1 = "@p"
   ->  Index Scan using tb_alter_recomp_1_pkey on tb_alter_recomp_1
-        Index Cond: (c1 = 0)
+        Index Cond: (c1 = $1)
```

Comment:
this looks like query plans for regular queries can be affected by this change, since it affects the planner's inputs. My concern is that this fix can lead to changes in query plans for any query involving a variable. In that case there should be a mechanism to stop variables from being handled in this new way.

Reply:
Previously, we were generating wrong result (affecting principal of correctness). And new way is correct way of handling dynamic parameter. Changes in query plan is expected -- previously all the param were being folded as const which was wrong. And with this change, it can definitely spoils plan in some cases but thats expected as we cant assume anything about values at planning time.